src/main/java/org/nebula/contrib/ngbatis/proxy/MapperProxy.java

```diff
@@ -243,8 +241,18 @@ public static ResultSet executeWithParameter(ClassModel cm, MethodModel mm, Stri
                 + "\n\t- result：{}",
             proxyClass, proxyMethod, localSessionSpace, autoSwitch, gql, paramsForTemplate, result);
       }
-      if (localSession != null && !ResultSetUtil.isSessionError(result)) {
-        ENV.getDispatcher().offer(localSession);
+      handleSession(dispatcher, localSession, result);
+    }
+  }
+
+  private static void handleSession(SessionDispatcher dispatcher,
+      LocalSession localSession, ResultSet result) {
+    if (localSession != null) {
+      boolean sessionError = ResultSetUtil.isSessionError(result);
+      if (sessionError) {
+        dispatcher.release(localSession);

```

Comment:
When session error, such as connection reset, release a session wound be a better practices.
Otherwise the dead session will still in line.